kinoml/modeling/homology.py

```diff
@@ -0,0 +1,272 @@
+        """
+
+        import requests
+        from io import StringIO
+
+        if kinase:
+            if backend == "uniprot":
+                sequence = KinaseDomainAminoAcidSequence.from_uniprot(identifier)
+            elif backend == "ncbi":
+                sequence = KinaseDomainAminoAcidSequence.from_ncbi(identifier)
+            else:
+                raise ValueError(
+                    'Backend "%s" not supported. Please choose from ["uniprot", "ncbi"]'
+                    % (backend)
+                )
+
+        else:
+            params = {"query": identifier, "format": "fasta"}
+            response = requests.get("http://www.uniprot.org/uniprot/", params)
+
+            sequence = response.text.split("\n", 1)[1].replace("\n", "")
+
+        return sequence
+
+    def get_alignment(
+        self, template_system, canonical_sequence, pdb_entry=False, window=15
+    ):
+        #  WIP
+        #  TODO write output to a logger
+        import tempfile
+        import requests
+        from modeller import alignment, log, environ, model
+        import numpy as np
+        import re
+
+        log.verbose()
+        env = environ()
+
+        pdb_id = template_system.metadata["id"]
+
+        if pdb_entry == True:
+            from appdirs import user_cache_dir
+
+            template_system.metadata["path"] = f"{user_cache_dir()}/{pdb_id}.pdb"
+
+            # TODO there is probably a better way to this, it's a
+            # repeat of ..core.proteins.ProteinStructure.from_name
+            url = f"https://files.rcsb.org/download/{pdb_id}.pdb"
+            response = requests.get(url)
+
+            with open(
+                template_system.metadata["path"], "wb"
+            ) as pdb_file:  # saving the pdb to cache
+                pdb_file.write(response.content)
+
+        env.io.atom_files_directory = [
+            template_system.metadata["path"].split(".")[0].split(pdb_id)[0]
+        ]
+
+        aln = alignment(env)
+        mdl = model(env)
+
+        # Read the whole template structure file, gets sequence automatically
+        code = template_system.metadata["id"]
+        mdl.read(file=code, model_segment=("FIRST:@", "END:"))
+
+        # Add the template sequence to the alignment
+        aln.append_model(mdl, align_codes=code, atom_files=code)
+
+        # add the canonical target sequence
+        aln.append_sequence(canonical_sequence, blank_single_chain=True)
+
+        # edit canonical target sequence keywords
+        aln[1].code = "target_seq"  #  TODO set target sequence name to be UniProt ID?
+
+        # align the sequences TODO should we use salign() instead?
+        aln.align()
+
+        with tempfile.NamedTemporaryFile(suffix=".ali") as temp_file:
+            aln.write(file=temp_file)
+            temp_file.seek(0)  # rewind the file for reading
+            # TODO add option to save original alignment
+
+            ali_lines = []
+            for line in temp_file.readlines():
```

Comment:
You can iterate line by line directly, without `readlines()`. This prevents a huge list from being loaded in memory if the file is super huge!

```suggestion
            for line in temp_file:
```